### PR TITLE
Add dry run option to simulate script

### PR DESCRIPTION
**Pull request by Ivan-Velickovic**
Pretty much the change from https://github.com/seL4/seL4_tools/pull/143, I find it quite useful from time to time so it would be nice for it to be upstreamed.

### CODE REVIEW

**Comment by kent-mcleod** [comment]
Sure, I'd also find this useful.  Currently you can achieve this with `-b echo` which sets the binary to echo instead of qemu, but this is too hard to remember...

File: cmake-tool/simulate_scripts/simulate.py [hunk 1]
```diff
@@ -80,23 +82,26 @@ def notice(message):
...
-        msg = "QEMU failed; resetting terminal in {d} seconds".format(d=delay) \
-            + "--interrupt to abort\n"
-        notice(msg)
-    else:
-        delay = 2  # in seconds
-
-    time.sleep(delay)
-
-    subprocess.call("tput reset", shell=True)
+        notice('GDB command: ' + qemu_gdbserver_command)
```
Comment (kent-mcleod) [inline cmake-tool/simulate_scripts/simulate.py:88]: Why are these prints changed too? `qemu_gdbserver_command` is already part of qemu command I think?

Comment (Ivan-Velickovic) [reply cmake-tool/simulate_scripts/simulate.py]: Yes you're right, my bad.

Comment (Ivan-Velickovic) [reply cmake-tool/simulate_scripts/simulate.py]: (I've discarded that change)

**Review by kent-mcleod — APPROVED** [review]

File: cmake-tool/simulate_scripts/simulate.py [hunk 2]
```diff
@@ -80,23 +82,25 @@ def notice(message):
...
-    notice(qemu_simulate_command)
+    notice('QEMU command: ' + qemu_simulate_command)
 
     if qemu_gdbserver_command != "":
-        notice('waiting for GDB on port 1234...')
+        if not args.dry_run:
+            notice('waiting for GDB on port 1234...')
 
-    qemu_status = subprocess.call(qemu_simulate_command, shell=True)
+    if not args.dry_run:
```
[the remark this answers is not in the document]
Comment (Ivan-Velickovic) [reply cmake-tool/simulate_scripts/simulate.py]: Done, I've done `exit` since `return` is not possible.

**Review by lsf37 — APPROVED** [review]
Nice, will merge when the tests have passed.

**Pull request merged**